src/olympia/shelves/tests/test_serializers.py

```diff
@@ -78,85 +79,69 @@ def setUp(self):
             api_settings.DEFAULT_VERSIONING_CLASS()
         )
         self.request.version = api_version
-        self.request.user = AnonymousUser()
-
-    def get_serializer(self, instance, **extra_context):
-        extra_context['request'] = self.request
-        return ShelfSerializer(instance=instance, context=extra_context)
+        self.request.user = user_factory()
 
     def serialize(self, instance, **extra_context):
+        self.request.query_params = dict(parse.parse_qsl(
+            instance.criteria))
+        extra_context['request'] = self.request
+        if instance.endpoint == 'collections':
+            extra_context['action'] = 'list'
+            extra_context['kwargs'] = {
+                'user_pk': self.request.user.pk,
+                'collection_slug': instance.criteria}
```

Comment:
what's this extra_context for collections used for?  I still only see `request['context']` being used in the serializer.

Reply:
argh, I wrote that the wrong way around - I should have written `context['request']`.  
My question was that you're passing `action`, and `kwargs` in as extra context (or just `context` as renamed now) but neither appear to be used in the serializer code itself, so I was wondering if that was an oversight or they were used in some hidden way (if so, please add a comment in the code explaining why they need to be passed in)

Reply:
Aah, sorry it was an oversight! I had included that when I was trying to resolve the errors when running the tests and before realizing I had not added a collection in `setUpTestData`. I didn't check if it was necessary when all the tests had finally passed.